pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,49 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_terminating_func(node: nodes.Call) -> bool:
+        """Detect call to exit(), quit(), os._exit(), or sys.exit()."""
+        if (
+            not isinstance(node.func, nodes.Attribute)
+            and not (
```

Comment:
Yeah but if you re assign `exit` to `another_exit` it won't catch anymore.

Reply:
I'm sorry, I don't understand your ask. Please give a specific code example or update it yourself. I think it's fine as is.

Reply:
It's the same remark than in https://github.com/PyCQA/pylint/pull/7520#discussion_r979138466 afaiu

Reply:
I'm going through the added test cases and trying to figure out how to translate them to words, but it may make more sense for you to look at them and provide a test case that would fail the current code.  As written,  the current code in fact works great for `another_exit` function, which is what this added [test](https://github.com/PyCQA/pylint/pull/7520/files) handles. I'm trying to minimize back and forth here but I'm really not understanding how this would not work?

Reply:
I think it's something like this : https://github.com/PyCQA/pylint/pull/7520/files#r987218501

Reply:
Aha! Thank you PIerre, that's 💯 the feedback I was looking for. That edge case hadn't occurred to me. Will add the test and fix it!